packages/core/src/shared/ui/nestedWizardPrompter.ts

```diff
@@ -0,0 +1,63 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Wizard, WizardOptions } from '../wizards/wizard'
+import { Prompter } from './prompter'
+import { WizardPrompter } from './wizardPrompter'
+import { createHash } from 'crypto'
+
+/**
+ * An abstract class that extends the base Wizard class plus the ability to
+ * use other wizard classes as prompters
+ */
+export abstract class NestedWizard<T> extends Wizard<T> {
```

Comment:
Do we actually need this class or coudl we just add `childWizards` (or whatever) to the main `Wizard` base class?

Reply:
I think we can move all the mapping and hashing logic from this class to the `Wizard` class. 

I just thought that having an abstract class NestedWizard is a bit more trtansparent give the fact that the wizard class is very complex already.

Reply:
Moved logic from `NestedWizard` class to `Wizard` and remove the former.

Reply:
> Moved logic from NestedWizard class to Wizard and remove the former.

Actually, in doing so, we create a circular dependency between `wizardPrompter`, `wizard`, `wizardPrompter`. 
```
../core/dist/src/shared/ui/wizardPrompter.js -> ../core/dist/src/shared/wizards/wizard.js -> ../core/dist/src/shared/ui/wizardPrompter.js
```
Reverted the change.